File: 724. Find Pivot Index/Solution1.java
//idea:pivor index的左边数字之和=右边数字之和
//     => 数字总和-nums[pivor index] = 左边数字之和x2


class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;
        int left = 0;
        
        for(int i=0; i<nums.length;i++){
            sum += nums[i];
        }
        
        for(int i=0; i<nums.length;i++){
            if(left*2 == (sum-nums[i])){
                return i;
            }
               left += nums[i];
        }
        return -1;
    }
}

File: 75. Sort Colors/solution1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        for (int max = nums.size() - 1; max > 0; max--) {
            bool swapped = false;
            for (int i = 0; i < max; i++) {
                int left = nums[i];
                int right = nums[i + 1];
                if (left > right) {
                    nums[i + 1] = left;
                    nums[i] = right;
                    swapped = true;
                }
            }
        if (!swapped) break;
        }
    }
};
